server/routes/overview/supportNeedsController.ts

```diff
@@ -15,20 +15,37 @@ export default class SupportNeedsController {
 
     const supportNeeds = await this.curiousService.getPrisonerSupportNeeds(prisonNumber, req.user.username)
     const prisonNamesById = await this.prisonService.getAllPrisonNamesById(req.user.username)
+    let atLeastOnePrisonHasSupportNeeds = false
 
     // Loop through the healthAndSupport needs array and update the prison name for each need
     if (supportNeeds.healthAndSupportNeeds) {
-      supportNeeds.healthAndSupportNeeds.map(async supportNeed => {
+      supportNeeds.healthAndSupportNeeds = supportNeeds.healthAndSupportNeeds.map(supportNeed => {
```

Comment:
I'm not sure this change is right / I'm not quite sure what it is doing / I think it's now trying to do more than the comment on line 20 suggests 
(though I think you are right with the removal of `async`; and I think there is a minor problem with the current code through the use of `map` but it doesnt actually do anything with the return!

How about this:
```
    let atLeastOnePrisonHasSupportNeeds = false

    if (supportNeeds.healthAndSupportNeeds) {
      // Work out if any of the HealthAndSupportNeeds records contains any assessment data
      atLeastOnePrisonHasSupportNeeds = supportNeeds.healthAndSupportNeeds.some(
        supportNeed =>
          supportNeed.rapidAssessmentDate ||
          supportNeed.inDepthAssessmentDate ||
          supportNeed.primaryLddAndHealthNeeds ||
          (supportNeed.additionalLddAndHealthNeeds && supportNeed.additionalLddAndHealthNeeds.length > 0),
      )

      // Loop through the healthAndSupport needs array and update the prison name for each need
      supportNeeds.healthAndSupportNeeds = supportNeeds.healthAndSupportNeeds.map(supportNeed => {
        const prison = prisonNamesById.get(supportNeed.prisonId)
        return {
          ...supportNeed,
          prisonName: prison,
        }
      })
    }
```

(I've not tested the above, you'll need to play with it to see if it works / does what you are trying to achieve etc)